temporal-sdk/src/test/java/io/temporal/activity/ActivityMethodOptionsTest.java

```diff
@@ -20,98 +20,135 @@
+    Assert.assertEquals(defaultOps, merged);
     // Assert options were overridden with method options
-    merged = ActivityOptions.newBuilder(options).mergeActivityOptions(methodOptions).build();
-    Assert.assertEquals(methodOptions, merged);
+    merged = ActivityOptions.newBuilder(defaultOps).mergeActivityOptions(methodOps1).build();
+    Assert.assertEquals(methodOps1, merged);
   }
 
   @Test
-  public void testActivityRetryOptionsChange() {
-    Map<String, ActivityOptions> activityMethodOptions =
-        new HashMap<String, ActivityOptions>() {
-          {
-            put("method1", methodOptions);
-          }
-        };
+  public void testActivityMethodOptions() {
+    testEnv.registerActivitiesImplementations(new ActivityImpl());
+    TestActivity activity =
+        testEnv.newActivityStub(TestActivity.class, defaultOps, perMethodOptionsMap);
 
-    // Test that Map<Method, ActivityOptions> was created
-    ActivityInvocationHandler invocationHandler =
-        (ActivityInvocationHandler)
-            ActivityInvocationHandler.newInstance(
-                TestActivity.class, options, activityMethodOptions, null);
-    Map<Method, ActivityOptions> methodToOptionsMap = invocationHandler.getActivityMethodOptions();
-    POJOActivityInterfaceMetadata activityMetadata =
-        POJOActivityInterfaceMetadata.newInstance(TestActivity.class);
+    // Check that options for method1 were merged.
+    Map<String, String> method1OpsValues = activity.method1();
+    Assert.assertEquals(
+        methodOps2.getHeartbeatTimeout(), Duration.parse(method1OpsValues.get("HeartbeatTimeout")));
+    Assert.assertEquals(
+        defaultOps.getScheduleToCloseTimeout(),
+        Duration.parse(method1OpsValues.get("ScheduleToCloseTimeout")));
+    Assert.assertEquals(
+        methodOps2.getStartToCloseTimeout(),
+        Duration.parse(method1OpsValues.get("StartToCloseTimeout")));
 
-    for (POJOActivityMethodMetadata methodMetadata : activityMetadata.getMethodsMetadata()) {
-      Method method = methodMetadata.getMethod();
-      if (method.getName().equals("method1")) {
-        Assert.assertEquals(methodOptions, methodToOptionsMap.get(method));
-      } else {
-        Assert.assertEquals(options, methodToOptionsMap.get(method));
-      }
-    }
+    // Check that options for method2 were default.
+    Map<String, String> method2OpsValues = activity.method2();
+    Assert.assertEquals(
+        defaultOps.getHeartbeatTimeout(), Duration.parse(method2OpsValues.get("HeartbeatTimeout")));
+    Assert.assertEquals(
+        defaultOps.getScheduleToCloseTimeout(),
+        Duration.parse(method2OpsValues.get("ScheduleToCloseTimeout")));
+    Assert.assertEquals(
+        defaultOps.getStartToCloseTimeout(),
+        Duration.parse(method2OpsValues.get("StartToCloseTimeout")));
   }
 
   @ActivityInterface
   public interface TestActivity {
 
     @ActivityMethod
-    void method1();
+    Map<String, String> method1();
```

Comment:
Change the activity methods to return Map<String, Duration> to remove toString and parse.